### ISSUE TITLE
Batch size in bytes

### ISSUE THREAD

**Issue by Flamefork** [gocql/gocql#1448]
Is there any way to calculate the byte size of the batch while building it?

### What did you do?

Trying to batch same-partition updates to speed up data ingestion the way sstableloader does (and claim 5x improvement).

### What did you expect to see?

Not that I've expected it, but a function to calculate the size of the batch to avoid hitting `batch_size_fail_threshold_in_kb` would be very helpful.

### What did you see instead?

Only Java versions of it :)
[BatchStatement:: requestSizeInBytes](https://github.com/dbraley/datastax/blob/eace672d4532e5533ca516fc7dc9a051489c76d4/driver-core/src/main/java/com/datastax/driver/core/BatchStatement.java#L205)
[RegularStatement:: requestSizeInBytes](https://github.com/dbraley/datastax/blob/eace672d4532e5533ca516fc7dc9a051489c76d4/driver-core/src/main/java/com/datastax/driver/core/RegularStatement.java#L196)

### What version of Cassandra are you using?

3.11.4

### What version of Gocql are you using?

v0.0.0-20190523124812-0680bfb96414

**Comment by deckarep** [comment]
I agree, this would be immensely useful to catch batches that violate this limit before sending off to Cassandra.

**Comment by OleksiienkoMykyta** [comment]
@joao-r-reis maybe it should be added to your API design(https://issues.apache.org/jira/browse/CASSGO-22)? It would be nice to have one in it.

**Comment by joao-r-reis** [comment]
I think this is fundamentally different from CASSGO-22 but it's a good feature request so let's keep this issue
